File: src/backend/models/auth_model.py
from __future__ import annotations

from ..credentials import (
    authenticate_admin,
    authenticate_user,
    get_all_admins,
    get_all_users,
    register_user,
    verify_admin_code,
)


def validate_admin_code(code: str) -> bool:
    return verify_admin_code(code)


def login_admin(email: str, password: str):
    return authenticate_admin(email, password)


def login_user(email: str, password: str):
    return authenticate_user(email, password)


def create_user(email: str, password: str, name: str):
    return register_user(email, password, name)


def list_users():
    return get_all_users()


def list_admins():
    return get_all_admins()

File: src/models/authModel.js
import { requestApiJson } from './api'

export function verifyAdminCode(code) {
  return requestApiJson('/auth/admin/code', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ code }),
  })
}

export function loginAdmin(email, password) {
  return requestApiJson('/auth/admin/login', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ email, password }),
  })
}

export function loginUser(email, password) {
  return requestApiJson('/auth/user/login', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ email, password }),
  })
}

export function registerUser(name, email, password) {
  return requestApiJson('/auth/user/register', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ name, email, password }),
  })
}
